demo/Titanic/domain_lib/JourneyLib.py

```diff
@@ -1,30 +1,52 @@
 #!/usr/bin/env python3
 
-from simulation.floating_object import FloatingObject
 from system.titanic import Titanic
 from simulation.iceberg import Iceberg
 from simulation.titanic_in_ocean import TitanicInOcean
 from simulation.journey import Journey
 from simulation.ocean import Ocean
 from robot.api.deco import keyword
 
-class JourneyLib():
+
+class JourneyLib:
     def __init__(self):
         self.ocean = Ocean()
+        self.journey = Journey(self.ocean)
 
-    @keyword("Spawn titanic at coordinate N${n} W${w} with a heading of ${heading} and at speed of ${speed} knots")
-    def spawn(self, n:int , w:int, heading:int, speed:int):
-        t = Titanic(speed, "east")
-        tio = TitanicInOcean(t, n, w, speed, "west")
+    @keyword("Spawn titanic at coordinate long ${n} lat ${w} heading ${heading} at speed of ${speed} knots")
+    def spawn_titanic(self, long: float, lat: float, heading: str, speed: float):
+        """
+        Spawns the titanic with given parameters
+        @param long: longitude of titanic
+        @param lat: latitude of titanic
+        @param heading: heading of the titanic (east, west, south, north)
+        @param speed: speed in knots of the titanic
+        """
+        headings = {
+            "north": 0,
+            "east": 90,
+            "south": 180,
+            "west": 270,
+        }
+        t = Titanic(speed, steering_direction=0)
+        tio = TitanicInOcean(t, long, lat, speed, headings[heading])
         self.ocean.floating_objects.append(tio)
-        
-    @keyword("Spawn iceberg at coordinate N${n} W${w}")
-    def spawn_iceberg(self, n:int, w:int):
-        iceberg = Iceberg(n, w)
+
+    @keyword("Spawn iceberg at coordinate long ${long} lat ${lat}")
+    def spawn_iceberg(self, long: float, lat: float):
+        """
+        Spawns an iceberg with given parameters
+        @param long: longitude of iceberg
+        @param lat: latitude of iceberg
+        """
+        iceberg = Iceberg(long, lat)
         self.ocean.floating_objects.append(iceberg)
 
-    @keyword("Pass time")
-    def pass_time(self):
-        j = Journey(self.ocean)
-        j.passed_time(10)
+    @keyword("play out Journey for a duration of ${minutes} minutes")
```

Comment:
Logging is very limited from this keyword. I think we need a bit more transparency to follow along what is happening.

Reply:
Agreed. Will for now add some statements about what is going on with the Titanic.